generators/client-2/templates/src/main/webapp/app/blocks/handlers/_state.handler.ts

```diff
@@ -1,29 +1,30 @@
-import * as angular from 'angular';
+import { VERSION, DEBUG_INFO_ENABLED } from '../../app.constants';
+import { Transition } from "ui-router-ng2";
+declare var SystemJS;
 
-import { VERSION } from '../../app.constants';
+StateHandler.$inject = ['$rootScope', '$transitions', <% if (enableTranslation) { %>'$translate', /*'<%=jhiPrefixCapitalized%>LanguageService',*/ 'TranslationHandler',<% } %> '$window',
+        'Auth', 'Principal', '$uiRouter', '$trace'];
 
-StateHandler.$inject = ['$rootScope', '$state', '$sessionStorage', <% if (enableTranslation) { %>'$translate', /*'<%=jhiPrefixCapitalized%>LanguageService',*/ 'TranslationHandler',<% } %> '$window',
-        'Auth', 'Principal'];
+export function StateHandler($rootScope, $transitions, <% if (enableTranslation) { %>$translate, /*<%=jhiPrefixCapitalized%>LanguageService,*/ TranslationHandler,<% } %> $window,
+    Auth, Principal, $uiRouter, $trace) {
+
+    if (DEBUG_INFO_ENABLED) {
+        $trace.enable('TRANSITION');
+        SystemJS.import('ui-router-visualizer').then(vis => vis.visualizer($uiRouter));
+    }
 
-export function StateHandler($rootScope, $state, $sessionStorage, <% if (enableTranslation) { %>$translate, /*<%=jhiPrefixCapitalized%>LanguageService,*/ TranslationHandler,<% } %> $window,
-    Auth, Principal) {
     return {
         initialize: initialize
     };
 
     function initialize() {
         $rootScope.VERSION = VERSION;
+        let deregistrationFns = [];
 
-        var stateChangeStart = $rootScope.$on('$stateChangeStart', function (event, toState, toStateParams, fromState) {
```

Comment:
`$stateChangeStart` and friends are deprecated. They're not included in 1.0.0 by default (although there is a polyfill to re-enable them).

Switched this logic to the far more powerful transition hooks.